Apps/WebClient/src/Server/Services/DependentService.cs

```diff
@@ -0,0 +1,103 @@
+        /// <param name="patientDelegate">The injected patient registry provider.</param>
+        /// <param name="userDelegateDelegate">The User Delegate delegate to interact with the DB.</param>
+        public DependentService(
+            ILogger<DependentService> logger,
+            IHttpContextAccessor httpAccessor,
+            IPatientDelegate patientDelegate,
+            IUserDelegateDelegate userDelegateDelegate)
+        {
+            this.logger = logger;
+            this.httpContextAccessor = httpAccessor;
+            this.patientDelegate = patientDelegate;
+            this.userDelegateDelegate = userDelegateDelegate;
+        }
+
+        /// <inheritdoc />
+        public RequestResult<DependentModel> AddDependent(string delegateHdId, AddDependentRequest addDependentRequest)
+        {
+            this.logger.LogTrace($"Dependent hdid: {delegateHdId}");
+            this.logger.LogDebug("Getting dependent details...");
+            string jwtString = this.httpContextAccessor.HttpContext.Request.Headers["Authorization"][0];
+            ResourceIdentifier identifier = new ResourceIdentifier("phn", addDependentRequest.PHN);
+            RequestResult<PatientModel> patientResult = this.patientDelegate.GetPatientByIdentifier(identifier, jwtString);
+            if (patientResult.ResourcePayload == null)
+            {
+                return new RequestResult<DependentModel>()
+                {
+                    ResultStatus = ResultType.Error,
+                    ResultError = new RequestResultError() { ResultMessage = "Communication Exception when trying to retrieve the Dependent", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.Patient) },
+                };
+            }
+
+            // Verify dependent's details entered by user
+            if (!patientResult.ResourcePayload.LastName.Equals(addDependentRequest.LastName, StringComparison.OrdinalIgnoreCase)
+                || !patientResult.ResourcePayload.FirstName.Equals(addDependentRequest.FirstName, StringComparison.OrdinalIgnoreCase)
+                || patientResult.ResourcePayload.Birthdate.Year != addDependentRequest.DateOfBirth.Year
+                || patientResult.ResourcePayload.Birthdate.Month != addDependentRequest.DateOfBirth.Month
+                || patientResult.ResourcePayload.Birthdate.Day != addDependentRequest.DateOfBirth.Day
+                || !patientResult.ResourcePayload.Gender.Equals(addDependentRequest.Gender, StringComparison.OrdinalIgnoreCase))
+            {
+                return new RequestResult<DependentModel>()
+                {
+                    ResultStatus = ResultType.Error,
+                    ResultError = new RequestResultError() { ResultMessage = "Information of the Dependent enterned don't match.", ErrorCode = ErrorTranslator.ServiceError(ErrorType.InvalidState, ServiceType.Patient) },
```

Comment:
grammer and typo - please get wording from Nino

Reply:
@ninosamson can you please provide the error message?

Reply:
per @ninosamson, the err message should be "The information you entered did not match. Please try again."